### DISCUSSION TITLE
How to make automapper to only query those columns that are really being mapped, to the database

### DISCUSSION THREAD

**Discussion by clopez30** [AutoMapper/AutoMapper.Extensions.OData#208] [Q&A] [answered]
Hi,
I am using the 'GetAsync' extension. It is working nicely but I have noticed that all columns are being queried and retrieved from the database, even if only a few are actually being mapped. the same occurs when only specific columns are requested using $select. I would expect that only requested columns are queried, but all they are again being selected in the generated SQL statement.

**Comment by BlaiseD** [comment, marked as answer]
`GetQueryAsync` should work.  `GetAsync` does not support that.

**Reply by clopez30** [reply]
It works. Thanks!
